src/renderers/shared/reconciler/ReactCompositeComponent.js

```diff
@@ -659,16 +660,16 @@ var ReactCompositeComponentMixin = {
         transaction,
         this._context
       );
-    }
-
-    if (this._pendingStateQueue !== null || this._pendingForceUpdate) {
+    } else if (this._pendingStateQueue !== null || this._pendingForceUpdate) {
       this.updateComponent(
         transaction,
         this._currentElement,
         this._currentElement,
         this._context,
         this._context
       );
+    } else {
+      this._updateBatchNumber = null;
```

Comment:
This line seems to imply that `enqueueUpdate` is getting called on a component that does not need to be updated (meets neither of the two conditions above) - which seems bad/wrong.  What am I missing here?